### PR TITLE
tor: update to version 0.4.1.6

### PR DESCRIPTION
**Pull request by svschmel**
Hello,

with PR 11695 i want to update tor to version 0.4.1.5.
On 2019-09-19 the Tor Project released the new version 0.4.1.6.
With this PR we can offer the new one.

New in this version:
- The tor binary will now be installed in the "/usr/bin" directory.
- In the old versions it was "/usr/lib"
- The script "tor.sh" from the manifest was adjusted.

On <https://metrics.torproject.org/rs.html#details/AD0FCFD83EA4899DB34735E599701592DEA13690> you can the see new version running.

With "arm" you can check connections made by tor.
See the attachments.

IMHO we can close the PR 11695 and use this one to put tor V. 0.4.1.6 to oi-userland .

Screenshot_20190921_205842
Screenshot_20190921_210012

### CODE REVIEW

File: components/network/tor/tor.p5m [hunk 1]
```diff
@@ -31,15 +33,20 @@ dir path=var/lib/tor group=daemon mode=0755 owner=daemon
 dir path=var/log/tor group=daemon mode=0755 owner=daemon
 
 file etc/tor/torrc.sample path=etc/tor/torrc mode=0644 preserve=true
-
-file usr/bin/tor path=usr/lib/tor mode=0555
+file usr/bin/tor path=usr/bin/tor mode=0555
```
Comment (Mno-hime) [inline components/network/tor/tor.p5m]: This should be now enough:
```suggestion
file usr/bin/tor
```

File: components/network/tor/manifests/sample-manifest.p5m [hunk 2]
```diff
@@ -25,13 +25,16 @@ license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
 file path=etc/tor/torrc.sample
```
Comment (Mno-hime) [inline components/network/tor/manifests/sample-manifest.p5m:25]: Should we deliver `etc/tor/torrc.sample`?

**Review by Mno-hime — COMMENTED** [review]
Please do a bit of a clean-up of the `Makefile`:
- move `REQUIRED_PACKAGES += text/gnu-diffutils` to the rest of `REQUIRED_PACKAGES`;
- remove the commented line `#CONFIGURE_OPTIONS+=	--enable-linker-hardening`;
- drop `prep`, `configure`, and `ips` includes and replace them with `common.mk` include, drop "common targets" `build`, `install`, and `test` and put `BUILD_BITS=64` before the first include on top of the file, `CONFIGURE_BINDIR.64 = $(USRBINDIR)` likely should be dropped.

When done, rune `gmake clean sample-manifest` and then `vimdiff tor.p5m manifest/sample-manifest.p5m` and check that nothing got broken and all files we likely want to deliver are delivered as part of the tor.p5m file.

Did you run the `test` target? Did it pass?

File: components/network/tor/tor.p5m (hunk 1 above)
Comment (svschmel) [reply components/network/tor/tor.p5m]: Agree and fixed.

File: components/network/tor/manifests/sample-manifest.p5m (hunk 2 above)
Comment (svschmel) [reply components/network/tor/manifests/sample-manifest.p5m:25]: IMHO it is better to have an example configuration file especially for those users which never have used tor before.
In the sample file many options are explained.

**Comment by svschmel** [comment]
Hello,
i made the changes in the Makefile and run "gmake clean sample-manifest"

vimdiff => The main difference is the following block:

sample-manifest.p5m:
file path=etc/tor/torrc.sample                                                        
file path=usr/bin/tor 

tor.p5m:
file files/auth_attr path=etc/security/auth_attr.d/tor
file files/prof_attr path=etc/security/prof_attr.d/tor
file files/tor.xml \
    path=lib/svc/manifest/network/tor.xml
file files/tor.sh path=/lib/svc/method/tor.sh mode=0555
dir path=var/lib/tor group=daemon mode=0755 owner=daemon
dir path=var/log/tor group=daemon mode=0755 owner=daemon
file etc/tor/torrc.sample path=etc/tor/torrc mode=0644 preserve=true
file usr/bin/tor path=usr/bin/tor mode=0555

It looks ok for me. I don't think, that the sample-manifest-file knows something about the service-manifest files. => files-directory.
And i can't find something about this files-directory in the make-rules.

```
------
test target:

One test failed:

============================================================================
Testsuite summary for tor 0.4.1.6
============================================================================
# TOTAL: 20
# PASS:  16
# SKIP:  3
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See ./test-suite.log
============================================================================

/oi-userland/components/network/tor/build/amd64$ grep -v OK test-suite.log 

....
FAIL: src/test/test
===================

util/load_win_lib: SKIPPED
util/log_mallinfo: SKIPPED
util/map_anon_nofork: 
  FAIL /export/home/svschmel/oi-userland/components/network/tor/tor-0.4.1.6/src/test/test_util.c:6223: assert(buf[0] OP_EQ 0xd0): -48 vs 208
  [map_anon_nofork FAILED]
1/1360 TESTS FAILED. (2 skipped)
FAIL src/test/test (exit status: 1)
....
```

At the moment i don't know, if this is critical because the build version is up and running.
But it seems that this test have problems in the 0.4.x series:
https://trac.torproject.org/projects/tor/ticket/29529
https://trac.torproject.org/projects/tor/ticket/29534

File: components/network/tor/manifests/sample-manifest.p5m (hunk 2 above)
Comment (Mno-hime) [reply components/network/tor/manifests/sample-manifest.p5m:25]: Good, let's deliver it then.

**Comment by Mno-hime** [comment]
> It looks ok for me. I don't think, that the sample-manifest-file knows something about the service-manifest files. => files-directory.
> And i can't find something about this files-directory in the make-rules.

Both are OK. Files from `files` directory are addressed directly, see lines in tor.p5m starting `file files/`.

> At the moment i don't know, if this is critical because the build version is up and running.
> But it seems that this test have problems in the 0.4.x series:
> https://trac.torproject.org/projects/tor/ticket/29529
> https://trac.torproject.org/projects/tor/ticket/29534

They eventually enabled it again (https://github.com/torproject/tor/blob/master/src/test/test_util.c#L6397), please file bug upstream and note it above the `test` target in the Makefile.

(Please, don't create a new PR, push to this one. Thanks!)

**Comment by svschmel** [comment]
For the "map_anon_nofork"- test error i created the following ticket:
https://trac.torproject.org/projects/tor/ticket/31897

**Comment by svschmel** [comment]
Hello,

i have updated the Makefile.
I mentioned the issue request at torproject.org

File: components/network/tor/tor.p5m (hunk 1 above)
Comment (Mno-hime) [reply components/network/tor/tor.p5m]: I am sorry, I mislead you here, it should be: `file path=usr/bin/tor`. Otherwise it won't publish.

Comment (Mno-hime) [reply components/network/tor/tor.p5m]: I fixed it in this PR.

**Review by Mno-hime — APPROVED** [review]

File: components/network/tor/Makefile [hunk 3]
```diff
@@ -63,12 +57,22 @@ build:		$(BUILD_64)
 
 install:	$(INSTALL_64)
 
+# One test failed:
+# error: util/map_anon_nofork:
+# FAIL ~/oi-userland/components/network/tor/tor-0.4.1.6/src/test/test_util.c:6223: assert(buf[0] OP_EQ 0xd0): -48 vs 208
+#  [map_anon_nofork FAILED]
+# see: https://trac.torproject.org/projects/tor/ticket/31897
 test:		$(TEST_64)
```
Comment (AndWac) [inline components/network/tor/Makefile]: IMO install: and test: targets aren't needed anymore because they are included in common.mk.

**Pull request merged**